documentation/Blob Structure.md

```diff
@@ -0,0 +1,106 @@
+
+- each supported compression algorithm could map to one of the "application ids" (so it's more of a data format identifier rather than an application identifier)
+    - in this way, multiple rollups can find and decompress the same compressed data. We will then need a way to split the decompressed data by rollups. I suggest using the same format inside the compressed segment.
+- his post allows for up to 1535 applications per blob. This has some cascading effects:
+    - the header is expected to be sorted by application ID, and each application will look for their record (which tells them where to find the actual data) using a binary search
+    - applications must use this search mechanism precisely. If the header is not sorted correctly or the binary search fails for whatever reason, the data should be considered missing (or junk). We do not want a scenario where different clients reach different conclusions about what data is stored for a given application ID
+    - even though each of our publications can span several blobs, I expect there will only be a handful of rollups per publication (and they may even be grouped under the same compression id). Instead of doing a binary search, I recommend applications do a linear search through the whole header to ensure it has the expected structure (ie. sorted application IDs)
+
+
+## The proposal
+
+With these modifications to Dankrad's proposal, here is a complete description of my suggestion. I will describe what an L2 rollup node shoud do to update it state. The corresponding protocol for proposers and provers should follow pretty naturally.
+
+### Preparation
+
+Whoever configures the rollup should publish (perhaps in an on-chain registry) the full list of unique 3-byte data identifiers that the rollup nodes should process. This should include an identifier for:
+
+- its own rollup transactions
+- any other rollup transactions it should process (eg. for embedded rollups)
+- any other data feeds it should process (eg. for synchrony or coordination)
+- any compression (or other data processing) algorithms it supports
+
+### Publication Hashes
+
+The rollup node should find relevant publication hashes in the `PublicationFeed` contract. This will include anything published by its inbox plus any other data feeds that it is watching. There may be multiple relevant publications that share the same blobs, and the rollup node will need to track which blobs it is processing for which reasons. In the simplest case it is just looking for its own rollup transactions inside publications made by its own inbox.
+
+### Publication Headers
+
+For each relevant hash, the publication is the concatenation of all the corresponding blobs. The rollup node should retrieve and validate the first 31-byte chunk, and interpret it as a header with the structure:
+
+- the version (set to zero) (1 byte)
+- header length in chunks, excluding this one (1 byte)
+    - I expect this to be zero most of the time
```

Comment:
Why header length is zero most of the time?

Reply:
because I suspect the 5 segments already described by the first chunk will be sufficient. We only need one segment per content type (eg. rollup, or compression mechanism). I expanded on this comment in the doc